FailedBonusActivationActivity/handler.ts

```diff
@@ -0,0 +1,197 @@
+  bonusActivation: BonusActivationWithFamilyUID
+});
+
+export const FailedBonusActivationSuccess = t.interface({
+  kind: t.literal("SUCCESS")
+});
+export type FailedBonusActivationSuccess = t.TypeOf<
+  typeof FailedBonusActivationSuccess
+>;
+
+export const InvalidInputFailure = t.interface({
+  kind: t.literal("INVALID_INPUT")
+});
+export type InvalidInputFailure = t.TypeOf<typeof InvalidInputFailure>;
+
+export const UnhandledFailure = t.interface({
+  kind: t.literal("UNHANDLED_FAILURE"),
+  reason: t.string
+});
+export type UnhandledFailure = t.TypeOf<typeof UnhandledFailure>;
+
+export const TransientFailure = t.interface({
+  kind: t.literal("TRANSIENT")
+});
+export type TransientFailure = t.TypeOf<typeof TransientFailure>;
+
+const FailedBonusActivationFailure = t.union(
+  [InvalidInputFailure, UnhandledFailure, TransientFailure],
+  "FailedBonusActivationFailure"
+);
+export type FailedBonusActivationFailure = t.TypeOf<
+  typeof FailedBonusActivationFailure
+>;
+
+export const FailedBonusActivationResult = t.taggedUnion("kind", [
+  FailedBonusActivationSuccess,
+  FailedBonusActivationFailure
+]);
+export type FailedBonusActivationResult = t.TypeOf<
+  typeof FailedBonusActivationResult
+>;
+
+type IFailedBonusActivationHandler = (
+  context: Context,
+  input: unknown
+) => Promise<FailedBonusActivationResult>;
+
+/**
+ * Converts a Promise<Either> into a TaskEither
+ * This is needed because our models return unconvenient type. Both left and rejection cases are handled as a TaskEither left
+ * @param lazyPromise a lazy promise to convert
+ *
+ * @returns either the query result or a query failure
+ */
+const fromQueryEither = <R>(
+  lazyPromise: () => Promise<Either<QueryError | Error, R>>
+): TaskEither<Error, R> =>
+  tryCatch(lazyPromise, toError).chain(errorOrResult =>
+    fromEither(errorOrResult).mapLeft(toError)
+  );
+
+/**
+ * Release the lock that was eventually acquired for this request. A release attempt on a lock that doesn't exist is considered successful.
+ *
+ * @param bonusLeaseModel an instance of BonusLeaseModel
+ * @param familyMembers the family of the requesting user
+ *
+ * @returns either a conflict error or the unique hash id of the family
+ */
+const relaseLockForUserFamily = (
+  bonusLeaseModel: BonusLeaseModel,
+  familyUID: FamilyUID
+): TaskEither<Error, void> => {
+  return fromQueryEither(() => bonusLeaseModel.deleteOneById(familyUID)).map(
+    _ => void 0
+  );
+};
+
+const updateBonusAsFailed = (
+  bonusActivationModel: BonusActivationModel,
+  bonusActivation: BonusActivationWithFamilyUID
+): TaskEither<Error, RetrievedBonusActivation> => {
+  return fromQueryEither(() => {
+    const bonusToUpdate: NewBonusActivation = {
+      ...bonusActivation,
+      id: bonusActivation.id as BonusCode & NonEmptyString,
+      kind: "INewBonusActivation",
+      status: BonusActivationStatusEnum.FAILED
+    };
+    return bonusActivationModel.createOrUpdate(
```

Comment:
this should be a simple update (not createOrUpdate) with a `RetrievedBonusActivation`